ChaosMod/Effects/db/Misc/MiscSpawnUfo.cpp

```diff
@@ -0,0 +1,334 @@
+/*
+	Author: Kanawanagasaki
+*/
+
+#include <algorithm>
+#include <math.h>
+#include <random>
+#include <stdafx.h>
+
+#include "Memory/Physics.h"
+#include "Util/Text.h"
+
+struct UfoData
+{
+	Object prop;
+	Ped target;
+	Vector3 destination = Vector3();
+	Vector3 prevDestination = Vector3();
+	Vector3 velocity = Vector3();
+	Vector3 acceleration = Vector3();
+	bool isElevating = false;
+	bool isRunningAway = false;
+	bool fadeOut = false;
+	int opacity = 255;
+};
+
+const float UFO_RADIUS = 26.f;
+const int MAX_UFO_AMOUNT = 3;
+const int SPAWN_INTERVAL = 13000;
+static Vector3 DROP_POS = Vector3(-1130.f, 4925.f, 350.f);
+
+static std::vector<UfoData> ufoList = {};
+static int prevTick = 0;
+static int prevSpawnTick = 0;
+static Camera ufoCamera = 0;
+
+static Ped GetPedForUfo()
+{
+	Ped player = PLAYER_PED_ID();
+
+	// 10% chance to select a player
+	if (g_Random.GetRandomInt(1, 10) == 1)
+	{
+		for (auto ufo : ufoList)
+			if (ufo.target == player)
+				return 0;
```

Comment:
```suggestion
		for (auto ufo : ufoList)
		{
			if (ufo.target == player)
			{
				return 0;
			}
		}
```
For the sake of consistency with the rest of the project